### PR TITLE
Dashboard fixes

### LINKED ISSUE #9345: Pressing `Delete` key when editing a data-link deletes the datalink itself
https://github.com/enso-org/enso/assets/1436948/ed28b5c4-d1ca-4a96-b2c4-93e122be42cf

As shown on the video.

### Actual behaviour

I'm editing settings of a datalink and I want to delete some characters _after_ my cursor - so instead of using Backspace, I press Delete. This causes the modal to hide and _deletes_ the datalink itself. Totally unexpected.

### Expected behaviour

When I'm editing contents of a text fields, the key strokes like Delete should not be passed to any other elements.

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Close https://github.com/enso-org/cloud-v2/issues/1037
  - Fix downloading files. The issue was that the S3 domain was not whitelisted.
- Fix #9345
- Fix https://github.com/enso-org/cloud-v2/issues/1012
  - Make "Update" button in assets panel (right panel) disabled until its value is updated.

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

File: app/ide-desktop/lib/dashboard/src/data/dataLinkSchema.json [hunk 1]
```diff
@@ -100,7 +100,7 @@
         "libraryName": { "const": "Standard.AWS" },
         "uri": {
           "title": "URI",
-          "description": "Must start with \"s3://\".",
+          "description": "Must start with \"s3://\". Folder name must have at least 3 characters.",
```
Comment (radeusgd) [inline app/ide-desktop/lib/dashboard/src/data/dataLinkSchema.json]:
```suggestion
          "description": "Must start with \"s3://\". Bucket name must have at least 3 characters.",
```

Comment (radeusgd) [reply app/ide-desktop/lib/dashboard/src/data/dataLinkSchema.json]:
```suggestion
          "description": "Must start with \"s3://\". The bucket name must have between 3 and 63 characters.",
```

I guess that'd make sense if we want to be precise.

I'm not sure if this is needed at all - the bucket name must firstly be a valid S3 bucket. So the requirements for what is a valid name and what not are clearly defined when someone _creates_ a bucket.

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/data/dataLinkSchema.json]: good point, i'll just revert it then. definitely better to need less text

**Review by MrFlashAccount — APPROVED** [review]

File: app/ide-desktop/lib/dashboard/src/utilities/event.ts [hunk 2]
```diff
@@ -50,3 +50,13 @@ export function isTextInputEvent(event: KeyboardEvent | React.KeyboardEvent) {
   // Allow `alt` key to be pressed in case it is being used to enter special characters.
   return !event.ctrlKey && !event.shiftKey && !event.metaKey && isTextInputKey(event)
 }
+
+/** Whether the element accepts keyboard events. */
```
Comment (MrFlashAccount) [inline app/ide-desktop/lib/dashboard/src/utilities/event.ts]: I think here it'd better to rephrase into `Whether the element accepts user input`, cause many elements accepts keyboard events, for example - select, button, a, many more :)

**Review by indiv0 — APPROVED** [review]
LGTM, but in the future, could we do 1 PR per fix? I realize these fixes are small but it makes it easier to review if each fix gets 1 PR. Easier to understand the changes.

**Pull request merged**